### ISSUE TITLE
support CTEs in CREATE statements

### ISSUE THREAD

**Issue by JLDLaughlin** [MaterializeInc/materialize#5092]
required for dbt support: https://github.com/MaterializeInc/materialize/issues/5086#issuecomment-747868915

Materialize should support `CREATE` statements like:
```
materialize=> create view my_view as (with my_cte as (select 1) select * from my_cte);
ERROR:  unknown catalog item 'my_cte'
```

**Comment by rjnn** [comment]
Can we add a release note (by user request)? not sure which release this landed in, 0.6.1 or 0.6.2.

**Comment by benesch** [comment]
Yeah, I'll do it when I do my release notes pass for v0.6.1.
